services/Donation.js

```diff
@@ -171,6 +171,52 @@ class DonationService {
       throw new ErrorHandler(e, 500);
     }
   }
+
+  /**
+   * @description Records a bulk donation not done via Flutterwave
+   * @param {object} data
+   */
+  static async recordBulkDonation(payload) {
+    const { name, email, phone_number, ...referralDetails } = payload;
+    try {
+      // This should mirror Flutterwave's webhook payload
+      const transactionPayload = {
+        tx_ref: referralDetails.tx_ref,
+        flw_ref: "Not applicable",
+        amount: referralDetails.amount,
+        currency: "NGN",
+        charged_amount: referralDetails.amount,
+        app_fee: 0,
+        merchant_fee: 0,
+        processor_response: "Successful",
+        auth_model: "INTERNET_BANKING",
+        status: "successful",
+        payment_type: "direct",
+        account_id: 1111111 // An arbitrary value just to mirror Flutterwave
+      };
+
+      const donorPayload = {
+        id: 1111111, // Arbitrary value to mirror FLW
+        name,
+        email,
+        phone_number
+      };
+
+      const donationData = {
+        transactionId: Math.floor(Math.random() * 9000000) + 1000000, // Arbitrary value,
```

Comment:
I would prefer we install the `uuid` [npm package](https://www.npmjs.com/package/uuid) and use that to generate the transaction id instead of this calculation.

Reply:
@edustipend-root The transaction id *must* be an integer. Strings are not allowed. (Unless I will also modify that part of the schema)

Reply:
Alright, @Chiemezuo. We can revisit that in a future PR.